libr/anal/vtable.c

```diff
@@ -90,53 +94,56 @@ static bool vtable_section_can_contain_vtables(RVTableContext *context, RBinSect
 		r_str_endswith (section->name, "__const");
 }
 
-
-static int vtable_is_addr_vtable_start(RVTableContext *context, ut64 curAddress) {
-	RAnalRef *xref;
-	RListIter *xrefIter;
+static int vtable_is_addr_vtable_start(RVTableContext *context, ut64 curAddress, ut64 data_section_start, ut64 data_section_end) {
+	ut64 value;
 
 	if (!curAddress || curAddress == UT64_MAX) {
 		return false;
 	}
 	if (curAddress && !vtable_is_value_in_text_section (context, curAddress, NULL)) {
 		return false;
 	}
-	// total xref's to curAddress
-	RList *xrefs = r_anal_xrefs_get (context->anal, curAddress);
-	if (r_list_empty (xrefs)) {
-		r_list_free (xrefs);
+
+	if (!context->read_addr (context->anal, curAddress - context->word_size, &value)) {
+		return false;
+	}
+	if (value != 0 && (value < data_section_start || value >= data_section_end)) {
+		return false;
+	}
+	if (!context->read_addr (context->anal, curAddress - 2 * context->word_size, &value)) {
+		return false;
+	}
+	if ((st32)value > 0) {
```

Comment:
This almost never holds with MSVC. I think it doesn't have a second value before the vtables at all. If we remove this condition, we get a lot of false positives unfortunately.

Reply:
ok i think the safest way to go is doing different checks for different ABIs.

To my perspective in general it's better having false positives than nothing at all at this level, though. Because false positives can be ruled out when trying to actually use the data produced here, while false negatives makes the user completely blind.